ext/dom/attr.c

```diff
@@ -50,14 +50,14 @@ PHP_METHOD(DOMAttr, __construct)
 	name_valid = xmlValidateName((xmlChar *) name, 0);
 	if (name_valid != 0) {
 		php_dom_throw_error(INVALID_CHARACTER_ERR, 1);
-		RETURN_FALSE;
+		RETURN_THROWS();
```

Comment:
as you wrote, `php_dom_throw_error()` does not always throw, so we can't be sure about the exception, thus the `RETURN_FALSE;` I think an RFC is needed to get rid of the flag that controls the error mode.

Reply:
It always throws when you pass ``1`` as the second argument, as this controls the throwing behaviour as can be seen here: https://heap.space/xref/php-src/ext/dom/domexception.c?r=dfd0acf0#35
So the ``RETURN_THROWS()`` in these cases is the valid return macro compared to ``RETURN_FALSE``

Reply:
Oops. I didn't pay attention to it. Thanks!